Update style card design

thomasguillot opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

To be in line with the new onboarding design, this PR updates the style of `StyleCard` and the Theme Wizard. 

__After:__

![1-after](https://user-images.githubusercontent.com/177929/104595560-af26a680-566a-11eb-8d45-a28e58850551.png)

__Before:__

![2-before](https://user-images.githubusercontent.com/177929/104595575-b2ba2d80-566a-11eb-96ca-e5d9b60d85cf.png)

### How to test the changes in this Pull Request:

1. Go to Site Design, check the old style
2. Switch to this branch and refresh wizard
3. Do you see the new style? Can you view demo/activate properly? 

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

dkoo reviewed (changes requested):
@thomasguillot This is looking good! One thing I noticed which is related but outside the scope of this PR is that the demo staging sites for each theme seem to lack any content. This kind of looks like a mistake because when clicking "View Demo" you are presented with an empty shell of a page rather than a sample homepage or single page. It would be great if the homepages for the demo staging sites had some content to showcase.

I also have a few suggestions which I've pushed to this branch as 8f0c42e. These are just suggestions, so feel free to push more changes or revert this commit, and I'll approve after your response. To summarize:

* The tab indexing doesn't quite work for keyboard-only navigation in the site design wizard. Removing the `tabindex` attribute from the container `div` and adding them to each button instead allows you to navigate every "Activate" and "View Demo" button without a mouse, and using `:focus-within` along with the hover styles lets the buttons stay visible if you've tabbed over the buttons instead of hovering.
* Now that we have the "quiet" loading style implemented in #798, I think this is more appropriate for ops like updating the active theme or saving settings.

thomasguillot commented:
@dkoo Both are great suggestions that I didn't think about, thank you!

I removed some unused CSS, added the demo button when the theme is active (you might still want to see the demo...) and I removed the image conditional check since to me, it doesn't make any sense to use this component without an image 😅

dkoo reviewed (approved):
LGTM! 👍
